Make awaiting `Join` trigger not return `Task` result (#3931)
cocotb/cocotb: docs/source/newsfragments/3931.change.rst

> Awaiting :meth:`Task.join() <cocotb.task.Task.join>` no longer returns the results of the :class:`~cocotb.task.Task`. To wait for the task to complete and yield the result of the task, await the task directly: ``await task``.

Reviewer: ```suggestion
Awaiting :meth:`Task.join() <cocotb.task.Task.join>` now returns the :class:`~cocotb.triggers.Join` trigger, not the result of the :class:`~cocotb.task.Task`. To wait for the task to complete and yield the result of the task, await the task directly: ``await task``.
```
